use-cases/meeting-scheduling/src/main/java/org/acme/meetingschedule/domain/MeetingAssignment.java

```diff
@@ -99,23 +105,30 @@ public int calculateOverlap(MeetingAssignment other) {
         // start is inclusive, end is exclusive
         int start = startingTimeGrain.getGrainIndex();
         int otherStart = other.startingTimeGrain.getGrainIndex();
-        int otherEnd = otherStart + other.meeting.getDurationInGrains();
-        if (otherEnd < start) {
+        if (other.getEndingTimeGrainDuration() < start) {
             return 0;
         }
-        int end = start + meeting.getDurationInGrains();
-        if (end < otherStart) {
+        if (getEndingTimeGrainDuration() < otherStart) {
             return 0;
         }
-        return Math.min(end, otherEnd) - Math.max(start, otherStart);
+        return Math.min(getEndingTimeGrainDuration(), other.getEndingTimeGrainDuration()) - Math.max(start, otherStart);
     }
 
     @JsonIgnore
-    public Integer getLastTimeGrainIndex() {
-        if (startingTimeGrain == null) {
+    public Integer getEndingTimeGrainIndex() {
+        int duration = getEndingTimeGrainDuration();
+        if (duration == 0) {
             return null;
         }
-        return startingTimeGrain.getGrainIndex() + meeting.getDurationInGrains() - 1;
+        return duration - 1;
+    }
+
+    @JsonIgnore
+    public int getEndingTimeGrainDuration() {
+        if (startingTimeGrain == null) {
+            return 0;
+        }
+        return startingTimeGrain.getGrainIndex() + meeting.getDurationInGrains();
     }
```

Comment:
Now this is confusing.

`lastTimeGrainIndex` makes perfect sense - it is the last index. But `endingTimeGrainDuration` is also a grain index; arguably it's not a duration at all, because it starts from zero, not from the starting time grain.

The difference between the two is effectively `-1`. And that, arguably, isn't worth the confusion, or worth us sitting down and figuring out a better name for the method. Therefore my proposal:

- Keep `lastTimeGrainIndex` as that's consistent with `firstTimeGrainIndex`.
- Make `endingTimeGrainDuration` be `lastTimeGrainIndex + 1`.
- Inline all uses of `endingTimeGrainDuration`, removing the method entirely. Can be done with IntelliJ refactoring tools.